inc/TRestGeant4PrimaryGeneratorInfo.h

```diff
@@ -0,0 +1,158 @@
+
+#ifndef REST_TRESTGEANT4PRIMARYGENERATORINFO_H
+#define REST_TRESTGEANT4PRIMARYGENERATORINFO_H
+
+#include <TF1.h>
+#include <TString.h>
+#include <TVector3.h>
+
+#include <string>
+
+namespace TRestGeant4PrimaryGeneratorTypes {
+
+enum class SpatialGeneratorTypes {
+    CUSTOM,
+    VOLUME,
+    SURFACE,
+    POINT,
+};
+
+std::string SpatialGeneratorTypesToString(const SpatialGeneratorTypes&);
+SpatialGeneratorTypes StringToSpatialGeneratorTypes(const std::string&);
+
+enum class SpatialGeneratorShapes {
+    GDML,
+    WALL,
+    CIRCLE,
+    BOX,
+    SPHERE,
+    CYLINDER,
+};
+
```

Comment:
`constexp std:map<SpatialGeneratorShapes, std::string> SpatialGeneratorTypesMap`

Reply:
I agree we should revisit this but I will do it in a separate PR, feel free to implement the changes if you want.